osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinnerTick.cs

```diff
@@ -25,6 +27,26 @@ public DrawableSpinnerTick(SpinnerTick spinnerTick)
             Origin = Anchor.Centre;
         }
 
+        protected override void OnApply()
+        {
+            base.OnApply();
+
+            // the tick can be theoretically judged at any point in the spinner's duration,
+            // so it must be alive throughout the spinner's entire lifetime.
+            // this mostly matters for correct sample playback.
+            LifetimeStart = DrawableSpinner.HitObject.StartTime;
+        }
+
+        protected override void UpdateHitStateTransforms(ArmedState state)
+        {
+            base.UpdateHitStateTransforms(state);
+
+            // the tick can be theoretically judged at any point in the spinner's duration,
+            // so it must be alive throughout the spinner's entire lifetime (or until hit, whichever applies).
+            // this mostly matters for correct sample playback.
+            LifetimeEnd = (Result?.IsHit == true ? Result.TimeAbsolute : DrawableSpinner.HitObject.GetEndTime()) + (Samples?.Length ?? 0);
+        }
```

Comment:
I think this can be removed, and the underlying logic in `DrawableHitObject` will handle things correctly (I'd like to hope it's engineered to handle these kinds of cases anyway):

https://github.com/ppy/osu/blob/a1a46e58fc09d8066d33ebbe98337fe1a33c3b16/osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs#L463-L464

In some quick testing it does look to be keeping things alive until they get judged.